src/theory/bv/theory_bv_rewrite_rules_normalization.h

```diff
@@ -1383,11 +1383,11 @@ bool RewriteRule<BitwiseSlicing>::applies(TNode node) {
       BitVector constant = node[i].getConst<BitVector>();
       // we do not apply the rule if the constant is all 0s or all 1s
       if (constant == BitVector(utils::getSize(node), 0u)) 
-        return false; 
-      
-      for (unsigned i = 0; i < constant.getSize(); ++i) {
-        if (!constant.isBitSet(i)) 
-          return true; 
+        return false;
+
+      for (unsigned j = 0, csize = constant.getSize(); j < csize; ++j)
+      {
+        if (!constant.isBitSet(j)) return true;
```

Comment:
Haha, good catch. Good thing is that the compiler does the correct thing here  (the two `i` are different), but it's still confusing.